src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.EcDsa.ImportExport.cs

```diff
@@ -0,0 +1,286 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using Microsoft.Win32.SafeHandles;
+using System;
+
+internal static partial class Interop
+{
+    internal static partial class Crypto
+    {
+        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_EcKeyGetCurveType")]
+        internal static extern ECCurve.ECCurveType EcKeyGetCurveType(SafeEcKeyHandle key);
+
+        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_EcKeyCreateByKeyParameters", CharSet = CharSet.Ansi)]
+        private static extern int EcKeyCreateByKeyParameters(
+            out SafeEcKeyHandle key,
+            string oid, 
+            byte[] qx, int qxLength, 
+            byte[] qy, int qyLength, 
+            byte[] d, int dLength);
+
+        internal static SafeEcKeyHandle EcKeyCreateByKeyParameters(
+            string oid,
+            byte[] qx, int qxLength,
+            byte[] qy, int qyLength,
+            byte[] d, int dLength)
+        {
+            SafeEcKeyHandle key;
+            int rc = EcKeyCreateByKeyParameters(out key, oid, qx, qxLength, qy, qyLength, d, dLength);
+            if (rc == -1)
+            {
+                throw new PlatformNotSupportedException(string.Format(SR.Cryptography_CurveNotSupported, oid));
```

Comment:
Nit: I expect key will still be non-null here, in which case we should Dispose of it to avoid it needing to be finalized (though obviously perf here isn't a big concern, given the exceptional path).